Helper/Cart.php

```diff
@@ -1807,4 +1850,38 @@ public function createCartByItem($item)
 
         return $cart_data;
     }
+
+    /**
+     * Assign customer to Quote by encrypted user id
+     *
+     * @param Quote $quote
+     * @param string $encrypted_user_id string generated by self::getEncodeUserId
+     */
+    private function assignQuoteCustomerByEncryptedUserId($quote, $encrypted_user_id)
+    {
+        $metadata = json_decode( $encrypted_user_id );
+        if ( ! $metadata || ! isset( $metadata->user_id ) || ! isset( $metadata->timestamp ) || ! isset( $metadata->signature ) ) {
+            throw new WebapiException(__('Incorrect encrypted_user_id'), 6306, 422);
```

Comment:
We are not doing a good job here not using "magic constants". I see that every other usage of `WebapiException` is using magic numbers here, so I won't block the PR, but I'm going to create a tech debt ticket in the M2 project to create constants for these error codes and missing HTTP codes.